src/@batch-flask/core/html-util/focus-helper.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Finds the first interactive element within an element tree and focuses it.
+ *
+ * @param element an HTML element tree to focus in
+ * @returns Whether an interactive was found and focused
+ */
+export function focusWithin(element: HTMLElement): boolean {
+    if (!element) {
+        return false;
+    }
+
+    if (isInteractive(element)) {
+        element.focus();
+        return true;
+    }
+
+    for (const child of element.childNodes) {
+        if (child instanceof HTMLElement && focusWithin(child)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+/**
+ * Determines if an HTML element is interactive.
+ *
+ * @param element An HTML element to query
+ * @returns whether the element is interactive
+ */
+export function isInteractive(element: HTMLElement): boolean {
```

Comment:
This should check if disabled="true" (disabled elements aren't focusable)

Reply:
This should probably check for `display: none` and `visibility: hidden` on the computed style.